### PR TITLE
pinegrow: bugfix for missing gsettings

### PR DESCRIPTION
**Pull request by gador**
The switch to makeWrapper was missing the gsettings directory. Pinegrow would start, but opening a project didn't work.
This fixes it.

Signed-off-by: Florian Brandes <florian.brandes@posteo.de>

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/editors/pinegrow/default.nix [hunk 1]
```diff
@@ -37,6 +37,8 @@ stdenv.mkDerivation rec {
   wrapProgramFlags = [
     "--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ gcc-unwrapped.lib gtk3 udev ]}"
     "--prefix PATH : ${lib.makeBinPath [ stdenv.cc ]}"
+    "--prefix XDG_DATA_DIRS : ${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}"
+    "--prefix XDG_DATA_DIRS : ${gtk3}/share/gsettings-schemas/${gtk3.name}"
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/pinegrow/default.nix:41]:
```suggestion
    "--prefix XDG_DATA_DIRS : ${gsettings-desktop-schemas}/share/gsettings-schemas/${gsettings-desktop-schemas.name}:${gtk3}/share/gsettings-schemas/${gtk3.name}"
```

**Comment by Artturin** [comment]
@gador i did some improvements, please test if it still wroks properly.

**Comment by gador** [comment]
@Artturin 
thanks for having a look!
Unfortunatly, your changes fail when trying to open a file:
`No GSettings schemas are installed on the system`

This is why I switched from `wrapGApps`. 

Maybe you have another idea?

**Comment by Artturin** [comment]
> @Artturin thanks for having a look! Unfortunatly, your changes fail when trying to open a file: `No GSettings schemas are installed on the system`
> 
> This is why I switched from `wrapGApps`.
> 
> Maybe you have another idea?

the setup hook that adds `GSETTINGS_SCHEMAS_PATH` which is used by `wrapGApps` is only set in the *fixup phases

thats why it wasn't adding the XDG_DATA_DIRS

**Comment by gador** [comment]
awesome! That did the trick!
If you want, I think it can be merged now.

**Pull request merged**